lib/deimos/utils/lag_reporter.rb

```diff
@@ -45,14 +45,26 @@ def compute_lag(topic, partition, offset)
           self.topics[topic.to_s] ||= Topic.new(topic, self)
           self.topics[topic.to_s].compute_lag(partition, offset)
         end
+
+        # Adjust the current lag based by asking Kafka based on the current offset.
+        # @param topic [String]
+        # @param partition [Integer]
+        # @param offset [Integer]
+        def recompute_lag(topic, partition)
+          self.topics[topic.to_s] ||= Topic.new(topic, self)
+          self.topics[topic.to_s].recompute_lag(partition)
+        end
       end
 
-      # Topic which has a hash of partition => last known offset lag
+      # Topic which has a hash of partition => last known offset lag and
+      # hash of partition => last known offset
       class Topic
         # @return [String]
         attr_accessor :topic_name
         # @return [Hash<Integer, Integer>]
         attr_accessor :partition_offset_lags
+        # @return [Hash<Integer, Integer>]
+        attr_accessor :partition_last_offsets
```

Comment:
This was mainly for keeping track of the last known offset to recalculate the lag based off of 
`current_last_offset - previous_last_offset` since the heartbeat doesn't provide the consumer groups current offset for the `topic:partition`

This could alternatively be recalculated by using the offset provided by the consumer's `seek` event

Either way, to recompute the lag I think we need either the current offset or the diff of the last known last offsets

Reply:
Maybe we should simplify this so that both `message_processed` and `seek` both just populate the current offset, and then `report_lag` will always get the last offset and do the simple subtraction. This seems way too complicated for a simple calculation. :)